randovania/bitpacking/construct_dataclass.py

```diff
@@ -129,6 +130,13 @@ def construct_for_type(type_: type) -> construct.Construct:
     elif dataclasses.is_dataclass(type_):
         return _construct_for_dataclass(type_)
 
+    elif hasattr(type_, "as_json") and hasattr(type_, "from_json"):
+        return construct.ExprAdapter(
+            BinStr,
+            encoder=lambda obj, ctx: json.dumps(obj),
+            decoder=lambda obj, ctx: json.loads(obj),
+        )
```

Comment:
is it a problem that building -> parsing the same object will transform it into JSON, rather than returning it to the class it previously was? I can't recall whether we enforce that here